isis/tests/FunctionalTestsCiss2isis.cpp

```diff
@@ -0,0 +1,273 @@
+
+  std::istringstream arss(R"(
+    Group = Archive
+      DataSetId     = CO-S-ISSNA/ISSWA-2-EDR-V1.0
+      ImageNumber   = 1472855646
+      ObservationId = ISS_00ASA_MOS0ASWE001_UVIS
+      ProductId     = 1_W1472855646.121
+    End_Group
+  )");
+
+  PvlGroup truthArchiveGroup;
+  arss >> truthArchiveGroup;
+
+  PvlGroup &archiveGroup = outLabel->findGroup("Archive", Pvl::Traverse);
+
+  EXPECT_PRED_FORMAT2(AssertPvlGroupEqual, archiveGroup, truthArchiveGroup);
+
+  std::istringstream bbss(R"(
+    Group = BandBin
+      FilterName   = CL1/CL2
+      OriginalBand = 1
+      Center       = 633.837
+      Width        = 285.938
+    End_Group
+  )");
+
+  PvlGroup truthBandBinGroup;
+  bbss >> truthBandBinGroup;
+
+  PvlGroup &bandBinGroup = outLabel->findGroup("BandBin", Pvl::Traverse);
+
+  EXPECT_PRED_FORMAT2(AssertPvlGroupEqual, bandBinGroup, truthBandBinGroup);
+
+  // Check for the ISS prefix pixel table
+  ASSERT_TRUE(outLabel->hasObject("Table"));
+  EXPECT_EQ(outLabel->findObject("Table", Pvl::Traverse)["Name"][0], "ISS Prefix Pixels");
+
+  std::unique_ptr<Histogram> hist (outCube.histogram());
+  EXPECT_NEAR(hist->Average(), 70.914941406249994, .00001);
+  EXPECT_EQ(hist->Sum(), 726169);
+  EXPECT_EQ(hist->ValidPixels(), 10240);
+  EXPECT_NEAR(hist->StandardDeviation(), 0.84419124016427105, .0001);
+}
+
+TEST(Ciss2Isis, Ciss2isisCustomMax) {
+  Pvl appLog;
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/ciss2isis_out.cub";
+
+  QString inputLabel = "data/ciss2isis/W1472855646_5.cropped.lbl";
+  QString updatedPvlLabel = prefix.path() + "/W1472855646_5.cropped.lbl";
+  Pvl inputPvl(inputLabel);
+  inputPvl["VALID_MAXIMUM"][1] = "70";
+  inputPvl.write(updatedPvlLabel);
+  QFile::copy("data/ciss2isis/W1472855646_5.cropped.img", prefix.path() + "/W1472855646_5.cropped.img");
+
+  QVector<QString> args = { "from=" + updatedPvlLabel,
+                            "to=" + cubeFileName };
+  UserInterface options(APP_XML, args);
+  try {
+    ciss2isis(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest file: " << e.toString().toStdString().c_str() << std::endl;
+  }
+
+  Cube outCube(cubeFileName);
+  Pvl *outLabel = outCube.label();
+
+  PvlGroup dimensions = outLabel->findGroup("Dimensions", Pvl::Traverse);
+  ASSERT_EQ((int)dimensions["Samples"], 1024);
+  ASSERT_EQ((int)dimensions["Lines"], 10);
+  ASSERT_EQ((int)dimensions["Bands"], 1);
+
+  PvlGroup pixels = outLabel->findGroup("Pixels", Pvl::Traverse);
+  ASSERT_EQ(pixels["Type"][0].toStdString(), "SignedWord");
+  ASSERT_EQ(pixels["ByteOrder"][0].toStdString(), "Lsb");
+  ASSERT_EQ((double)pixels["Base"], 0.0);
+  ASSERT_EQ((double)pixels["Multiplier"], 1.0);
+
+  std::unique_ptr<Histogram> hist (outCube.histogram());
+  EXPECT_NEAR(hist->Average(), 68.998626373626379, .00001);
+  EXPECT_EQ(hist->Sum(), 50231);
+  EXPECT_EQ(hist->ValidPixels(), 728);
+  EXPECT_NEAR(hist->StandardDeviation(), 0.037062465833055058, .0001);
```

Comment:
I don't think you need to check the average, sum, and standard deviation. I'd instead just check that the maximum is <= 70. I'd also check the number of HRS pixels as that's what things over the valid max are set to.

Reply:
So something interesting about the import app is that the max is also considered to be a fully saturated pixel. So all valid pixels if max == 70 will be < 70. This was something I discussed with @scsides a week or two ago and we decided that it is better to retain current functionality at this point